src/GameData.ts

```diff
@@ -156,9 +195,63 @@ export default class GameData {
         if (typeof result === "string") {
             message.channel.send(result);
             return;
+        } else {
+            message.channel.send(this.buildEmbed(result));
+        }
+    }
+
+    public buildEmbed(rawData: EmbeddableDatum) {
+        const embed = new Discord.RichEmbed();
+
+        switch (rawData.type) {
+            case "ChampionData": {
+                const imageString = `https://raw.communitydragon.org/latest/plugins/rcp-be-lol-game-data/global/default/v1/champion-icons/${rawData.id}.png`;
+                embed.setThumbnail(imageString);
+                embed.setURL(this.sharedSettings.lookup.championUrl);
+                break;
+            }
+            case "ItemData": {
+                const imageString = `https://raw.communitydragon.org/latest/plugins${(rawData as ItemData).iconPath}`
+                    .replace("lol-game-data", "rcp-be-lol-game-data/global/default")
+                    .replace("/assets", "")
+                    .toLowerCase();
+                embed.setThumbnail(imageString);
+                embed.setURL(this.sharedSettings.lookup.itemUrl);
+                break;
+            }
+            case "PerkData": {
+                const imageString = `https://raw.communitydragon.org/latest/plugins${(rawData as PerkData).iconPath}`
+                    .replace("lol-game-data", "rcp-be-lol-game-data/global/default")
+                    .replace("/assets", "")
+                    .toLowerCase();
+                embed.setThumbnail(imageString);
+                embed.setURL(this.sharedSettings.lookup.perkUrl);
+                break;
+            }
+        }
+        delete rawData.type;
+        if ("iconPath" in rawData) {
```

Comment:
I'm assuming typescript makes you do this? If not, you can lose the if (`delete` is a noop if the field doesn't exist).

Reply:
Yeah, TS is making me do this.